v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlex.java

```diff
@@ -156,24 +152,31 @@ public static void main(String[] args) throws IOException, RestClientException {
   public static PipelineResult run(KafkaToBigQueryFlexOptions options)
       throws IOException, RestClientException {
 
+    // Enable Streaming Engine
+    options.setEnableStreamingEngine(true);
+
+    // Enable Resource based billing
+    options.setDataflowServiceOptions(List.of("enable_streaming_engine_resource_based_billing"));
+
     // Validate BQ STORAGE_WRITE_API options
+    options.setUseStorageWriteApi(true);
+    if (options.getStorageWriteApiTriggeringFrequencySec() == null) {
+      options.setStorageWriteApiTriggeringFrequencySec(5);
+    }
     BigQueryIOUtils.validateBQStorageApiOptionsStreaming(options);
     MetadataValidator.validate(options);
 
     // Create the pipeline
     Pipeline pipeline = Pipeline.create(options);
 
     List<String> topicsList;
-    if (options.getKafkaReadTopics() != null) {
-      topicsList = new ArrayList<>(Arrays.asList(options.getKafkaReadTopics().split(",")));
-    } else {
-      throw new IllegalArgumentException("Please Provide --kafkaReadTopic");
-    }
     String bootstrapServers;
-    if (options.getReadBootstrapServers() != null) {
-      bootstrapServers = options.getReadBootstrapServers();
+    if (options.getReadBootstrapServerAndTopic() != null) {
```

Comment:
getReadBootstrapServerAndTopic() can be a GMK topic too, right?
Should we check if this matches the GMK topic pattern first, and only then try to parse it as <bootstrap_server>:<topic>?

Reply:
I was under the assumption that both GMK and Kafka would yield similar results `<bootstrap_server>:topic`

Reply:
Changed the logic based on the output.